sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -277,6 +278,31 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
     result.copy(output = newOutput)
   }
 
+  private def getDirectoryPathSeq(rootPath: Path): Seq[String] = {
+    val enableSupportSubDirectories =
+      sparkSession.conf.getOption("hive.mapred.supports.subdirectories")
```

Comment:
I don't think "hive.mapred.supports.subdirectories" should be a new Spark config. 
But it should be read from the hadoop configuration so use `hadoopConf.get(<>)` here.

Reply:
got it.

Reply:
should we create a new configuration for it?